src/rsa/bigint.rs

```diff
@@ -132,6 +132,18 @@ impl Positive {
         let bits = unsafe { GFp_BN_num_bits(self.as_ref()) };
         bits::BitLength::from_usize_bits(bits)
     }
+
+    // TODO Does this need to be constant time?
+    pub fn subtract_smaller(&self, smaller: &Positive) -> Result<Positive, error::Unspecified> {
```

Comment:
Also, since we know that `p` > `q` already, `elem_sub(copy_of_p, &q, &n)` is guaranteed to not to need any reduction because there won't be any underflow.